packages/dcos-integration-test/extra/resiliency/test_aws_cf.py

```diff
@@ -32,21 +33,20 @@ def dcos_launchpad(cluster):
 @pytest.mark.last
 def test_agent_failure(dcos_launchpad, cluster, vip_apps):
     # make sure the app works before starting
+    def get_running_agents(group_name):
```

Comment:
I think for now / this pr just adding the decorator is good enough.

Practically I _think_ when you construct the instance object with the instance ID the api call is made, and then .instance_id is accessing the local cache variable (although we'd have to verify).

Long term definitely we want to wrap boto clients directly. I _think_ http://boto3.readthedocs.io/en/latest/guide/events.html provides an event we could use (Creating client class, creating resource class) to wrap the objects, or maybe even hook into overriding the retry behavior. But I think we can safely enough postpone that until later.